File: packages/runtime/nodejs/src/workflow/type/state/state.ts
import {
  IFlowConstantRefValue,
  IFlowRefValue,
  WorkflowVariableType,
} from '@flowgram.ai/runtime-interface';

import { INode } from '../document';
import { WorkflowInputs, WorkflowOutputs } from '../base';
import { IVariableParseResult, IVariableStore } from './variable';

export interface IState {
  id: string;
  variableStore: IVariableStore;
  workflowInputs: WorkflowInputs;
  workflowOutputs: WorkflowOutputs;
  init(inputs: WorkflowInputs): void;
  dispose(): void;
  getNodeInputs(node: INode): WorkflowInputs;
  setNodeOutputs(params: { node: INode; outputs: WorkflowOutputs }): void;
  setWorkflowInputs(inputs: WorkflowInputs): void;
  setWorkflowOutputs(outputs: WorkflowOutputs): void;
  parseRef<T = unknown>(ref: IFlowRefValue): IVariableParseResult<T> | null;
  parseValue<T = unknown>(
    flowValue: IFlowConstantRefValue,
    type?: WorkflowVariableType
  ): IVariableParseResult<T> | null;
  isExecutedNode(node: INode): boolean;
  addExecutedNode(node: INode): void;
}

File: packages/runtime-go/interface/runtime/state.go
/**
 * Copyright (c) 2025 Bytedance Ltd. and/or its affiliates
 * SPDX-License-Identifier: MIT
 */

package runtimeType

import "flowgram-runtime-go/interface/schema"

type IState interface {
	GetID() string
	GetVariableStore() IVariableStore
	Init(schema *schema.WorkflowSchema)
	Dispose()
	GetNodeInputs(node INode[any]) WorkflowInputs
	SetNodeOutputs(node INode[any], outputs WorkflowOutputs)
	ParseInputs(values map[string]schema.IFlowValue, declare schema.IJsonSchema) WorkflowInputs
	ParseRef(ref schema.IFlowRefValue) *IVariableParseResult
	ParseTemplate(template schema.IFlowTemplateValue) *IVariableParseResult
	ParseFlowValue(flowValue schema.IFlowValue, declareType schema.WorkflowVariableType) *IVariableParseResult
	IsExecutedNode(node INode[any]) bool
	AddExecutedNode(node INode[any])
}
